### PR TITLE
test(grpc-suite): add post-upgrade gRPC surface coverage

### LINKED ISSUE #636: Automated post-upgrade gRPC transaction and query test suite
## Why

Network upgrades currently have no automated verification that the chain's full API surface still behaves correctly after the upgrade completes. An API regression found after an upgrade ships requires another network upgrade to fix. CLI issues, by contrast, can be fixed and released out-of-band — so CLI coverage is explicitly out of scope; the gRPC API is the surface that must be proven accurate.

## What

As part of the upgrade flow on a testnetify-forked network, automatically run a test suite post-upgrade that exercises:

1. Every transaction type the chain supports, including all possible edge cases.
2. Every query, including all possible edge cases.

All checks run against the gRPC API. The suite passing means the upgraded chain's API surface is verified accurate.

## Acceptance Criteria

- [X] Test suite runs automatically after an upgrade completes on a testnetify network.
- [ ] Every transaction type is exercised via gRPC, including all possible edge cases.
- [ ] Every query is exercised via gRPC, including all possible edge cases.
- [X] CLI is not part of the suite's scope.

[test(grpc-suite): add post-upgrade gRPC surface coverage by chalabi2 · Pull Request #2075 · akash-network/node](https://github.com/akash-network/node/pull/2075)

### PR DESCRIPTION
**Pull request by chalabi2**
## Description

Closes: [636](https://github.com/akash-network/support/issues/636)

Adds the post-upgrade gRPC surface suite for testnetify-forked upgrade runs.
The suite exercises the mounted Akash and Cosmos SDK gRPC transaction/query
surface after an upgrade and fails when discovered in-scope RPCs are not covered.

Key pieces:
- `tests/upgrade/grpcsuite`: authored tx packs, query smoke sweep, coverage gate
- `tests/upgrade/grpcsurface_worker_test.go`: universal post-upgrade worker
- `tests/fullsurface`: fast in-process local/CI driver
- `make test-grpc-surface[-tx|-query]`: local runners

Validation:
- `make -C tests/upgrade test`: PASS, coverage `tx 76/76`, `query 130/130`
- `make test-grpc-surface`: PASS, coverage `tx 76/76`, `query 130/130`
- `make test-grpc-surface-tx`: PASS, coverage `tx 76/76`
- `make test-grpc-surface-query`: PASS, coverage `query 130/130`

### CODE REVIEW

File: script/upgrades.sh [hunk 1]
```diff
@@ -520,8 +529,69 @@ function prepare_state() {
 
 	rvaldir=$validators_dir/.akash0
 
+	if [[ -n "$UPGRADE_TO" && -x "$rvaldir/cosmovisor/upgrades/$UPGRADE_TO/bin/akash" ]]; then
+		echo "rolling snapshot back before testnetify"
+		"$rvaldir/cosmovisor/upgrades/$UPGRADE_TO/bin/akash" rollback --home "$rvaldir" --hard
+		"$rvaldir/cosmovisor/upgrades/$UPGRADE_TO/bin/akash" rollback --home "$rvaldir" --hard
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply script/upgrades.sh:538]: Kept the second rollback intentionally and added an explanatory comment. Cached snapshots can sit at the last pre-upgrade height; testnetify needs to start one block earlier so it can commit fork-state changes before hitting the scheduled upgrade height.
